lgc/util/PassManager.cpp

```diff
@@ -108,25 +132,87 @@ static AnalysisID getPassIdFromName(StringRef passName) {
 }
 
 // =====================================================================================================================
-// Create a PassManagerImpl
+// Create a LegacyPassManagerImpl
 lgc::LegacyPassManager *lgc::LegacyPassManager::Create() {
+  return new LegacyPassManagerImpl;
+}
+
+// =====================================================================================================================
+// Create a PassManagerImpl
+lgc::PassManager *lgc::PassManager::Create() {
   return new PassManagerImpl;
 }
 
 // =====================================================================================================================
-PassManagerImpl::PassManagerImpl() : LegacyPassManager() {
+LegacyPassManagerImpl::LegacyPassManagerImpl() : LegacyPassManager() {
   if (!cl::DumpCfgAfter.empty())
     m_dumpCfgAfter = getPassIdFromName(cl::DumpCfgAfter);
 
   m_jumpThreading = getPassIdFromName("jump-threading");
   m_printModule = getPassIdFromName("print-module");
 }
 
+// =====================================================================================================================
+PassManagerImpl::PassManagerImpl() : PassManager(), instrVerify(getLgcOuts()) {
+  if (!cl::DumpCfgAfter.empty()) {
+    // TODO: We need to support m_dumpCfgAfter in a way that is similar to what
+    // is done in the add() function of the legacy pass manager. We can use the
+    // getPassNameForClassName function of PassInstrumentationCallbacks to get
+    // a pass ID from its name but we need to provide a mechanism to register
+    // custom passes there too.
+    report_fatal_error("-dump-cfg-after cannot be used with the new pass manager yet.");
+  }
+  // Setup custom instrumentation callbacks and register LLVM's default module
+  // analyses to the analysis manager.
+  registerCallbacks();
+  if (cl::VerifyIr)
+    instrVerify.registerCallbacks(instrCallbacks);
+  PassBuilder passBuilder(nullptr, PipelineTuningOptions(), None, &instrCallbacks);
+  passBuilder.registerModuleAnalyses(analysisManager);
+}
+
+// =====================================================================================================================
```

Comment:
Please add comments for this function and for the parameter `module`. This is required by the coding style even though they are trivial to do.